src/network/networkd-conf.c

```diff
@@ -62,27 +62,45 @@ int config_parse_dhcp_duid_raw(
         long byte;
         char *cbyte, *pnext;
         const char *pduid = (const char *)rvalue;
-        size_t count = 0, duid_len = 0;
-        Manager *m = userdata;
+        size_t count = 0, duid_index = 0;
+        Manager *m;
+        Network *n;
+        DHCPDUIDType duid_type;
+        size_t *duid_len;
+        struct duid *duid;
 
         assert(filename);
         assert(lvalue);
         assert(rvalue);
-        assert(m);
-        assert(m->dhcp_duid_type != _DHCP_DUID_TYPE_INVALID);
+        assert(userdata);
+
+        if (ltype == DUID_CONFIG_SOURCE_GLOBAL) {
+                m = userdata;
+                duid_type = m->dhcp_duid_type;
+                duid_len = &m->dhcp_duid_len;
+                duid = &m->dhcp_duid;
+        } else {
+                /* DUID_CONFIG_SOURCE_NETWORK */
+                n = userdata;
+                duid_type = n->dhcp_duid_type;
+                duid_len = &n->dhcp_duid_len;
+                duid = &n->dhcp_duid;
+        }
```

Comment:
This somehow seems overly complicated... You already take the offset to `struct duid` in .gperf, isn't this enough?